### PR TITLE
Add attribute_group type to share attributes across signals

### LINKED ISSUE #123: [Feature proposal] Support 'attribute group' type in semantic conventions
Attributes can appear on different signals (metrics, traces, events, links, logs), but we don't have a way to describe attributes without the signal in semantic conventions.
I'd like to have a mechanism, where I'd define attributes separately and then refer to them from spans/links/events definitions.

E.g. in messaging we can have attributes set on links or on spans, maybe events:

```yml
  - id: messaging.message
    prefix: messaging
    type: attribute_group # this is the proposal. We currently use 'span' in cases like this
    brief: 'Semantic convention describing per-message attributes populated on messaging spans or links.'
    attributes:
      - id: message.id
        type: string
        brief: message id
        note:  when sending individual messages, it should be populated on span, otherwise should be populated on links

  - id: messaging.publish
    prefix: messaging
    type: span
    brief:  ...
    attributes:
      - ref: messaging.message.id

  - id: messaging.link
    prefix: messaging
    type: link # out of scope of this proposal
    brief:  ...
    attributes:
      - ref: mesasging.message.id
```

We'd also benefit from it in [span-general.yaml](https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/general.yaml) which essentially describes unrelated groups of attributes, but not a span per se.

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #123

### CODE REVIEW

File: semantic-conventions/CHANGELOG.md [hunk 1]
```diff
@@ -4,7 +4,8 @@ Please update the changelog as part of any significant pull request.
 
 ## Unreleased
 
-- ...
+- Add a semantic convention type for generic attribute group ("attribute_group")
+  ([#TODO](https://github.com/open-telemetry/build-tools/pull/TODO)).
```
Comment (arminru) [inline semantic-conventions/CHANGELOG.md:8]:
```suggestion
  ([#124](https://github.com/open-telemetry/build-tools/pull/124)).
```

File: semantic-conventions/syntax.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ convtype ::= "span" # Default if not specified
          |   "event"    # see eventspecificfields
          |   "metric"   # (currently non-functional)
          |   "scope"
+         |   "attribute_group"
```
Comment (arminru) [inline semantic-conventions/syntax.md:52]: Please also add a section below documenting what it's used for and which fields it defines.

File: semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py [hunk 3]
```diff
@@ -546,5 +548,6 @@ def attributes(self):
         MetricSemanticConvention,
         UnitSemanticConvention,
         ScopeSemanticConvention,
+        AttributeGroupConvention
```
Comment (arminru) [inline semantic-conventions/src/opentelemetry/semconv/model/semantic_convention.py:551]:
```suggestion
        AttributeGroupConvention,
```
see https://github.com/open-telemetry/build-tools/actions/runs/3433329456/jobs/5723545460

File: semantic-conventions/src/tests/semconv/templating/test_markdown.py [hunk 4]
```diff
@@ -128,6 +128,9 @@ def testSamplingRelevant(self):
 
     def test_scope(self):
         self.check("markdown/scope/")
+
```
Comment (arminru) [inline semantic-conventions/src/tests/semconv/templating/test_markdown.py:131]:
```suggestion

```
see https://github.com/open-telemetry/build-tools/actions/runs/3433329456/jobs/5723545460

File: semantic-conventions/syntax.md (hunk 2 above)
Comment (lmolkova) [reply semantic-conventions/syntax.md]: @arminru I've added the description, can you please take another look? thanks!

Comment (arminru) [reply semantic-conventions/syntax.md]: Thanks!

File: semantic-conventions/syntax.md [hunk 5]
```diff
@@ -168,6 +169,12 @@ The following is only valid if `type` is `event`:
   If not specified, the `prefix` is used. If `prefix` is empty (or unspecified),
   `name` is required.
 
+#### Attribute group semantic convention
+
+Attribute group (`attribute_group` type) defines a set of attributes that can be 
+declared once and referenced by semantic conventions for different signals, for example spans and logs.
+Attribute groups don't have specific fields and follow general `semconv` semantics.
```
Comment (arminru) [inline semantic-conventions/syntax.md:176]:
```suggestion
Attribute groups don't have any specific fields and follow the general `semconv` semantics.
```

**Review by arminru — APPROVED** [review]

**Comment by lmolkova** [comment]
@open-telemetry/technical-committee can someone please merge? Thanks!

**Pull request merged**